commands/format.py

```diff
@@ -0,0 +1,110 @@
+import sublime_plugin
+
+from ..libs.viewhelpers import *
+from ..libs.texthelpers import *
+from ..libs.reference import *
+
+
+class TypescriptFormatOnKey(sublime_plugin.TextCommand):
+    """
+    Format on ";", "}", or "\n"; called by typing these keys in a ts file
+    in the case of "\n", this is only called when no completion dialogue visible
+    """
+    def is_enabled(self):
```

Comment:
For each command we need to define an is_enabled function, but I can see it is indeed clearer if we put that in a base class and then inherit from it. I'll try that.